c-scape/src/fs/stat.rs

```diff
@@ -0,0 +1,153 @@
+use rustix::fd::BorrowedFd;
+use rustix::ffi::ZStr;
+use rustix::fs::AtFlags;
+
+use core::convert::TryInto;
+use errno::{set_errno, Errno};
+use libc::{c_char, c_int, c_long, time_t};
+
+use crate::convert_res;
+
+fn rustix_stat_to_libc_stat(
+    rustix_stat: rustix::fs::Stat,
+) -> Result<libc::stat, std::num::TryFromIntError> {
+    // SAFETY: libc structs can be zero-initalized freely
+    let mut stat: libc::stat = unsafe { std::mem::zeroed() };
+    stat.st_dev = rustix_stat.st_dev as libc::dev_t;
+    stat.st_ino = rustix_stat.st_ino.try_into()?;
+    stat.st_nlink = rustix_stat.st_nlink as libc::nlink_t;
+    stat.st_mode = rustix_stat.st_mode as libc::mode_t;
+    stat.st_uid = rustix_stat.st_uid as libc::uid_t;
+    stat.st_gid = rustix_stat.st_gid as libc::gid_t;
+    stat.st_rdev = rustix_stat.st_rdev as libc::dev_t;
+    stat.st_size = rustix_stat.st_size.try_into()?;
+    stat.st_blksize = rustix_stat.st_blksize as libc::blksize_t;
+    stat.st_blocks = rustix_stat.st_blocks.try_into()?;
+    stat.st_atime = rustix_stat.st_atime as time_t;
+    stat.st_atime_nsec = rustix_stat.st_atime_nsec as c_long;
+    stat.st_mtime = rustix_stat.st_mtime as time_t;
+    stat.st_mtime_nsec = rustix_stat.st_mtime_nsec as c_long;
+    stat.st_ctime = rustix_stat.st_ctime as time_t;
+    stat.st_ctime_nsec = rustix_stat.st_ctime_nsec as c_long;
```

Comment:
For all these `as` casts, are we sure they won't overflow? It's tempting to make them all `.try_into()?`, and let the compiler optimize out the ones where the types are actually the same :-).